BrownianMotion/StochasticIntegral/ApproxSeq.lean

```diff
@@ -0,0 +1,129 @@
+/-
+Copyright (c) 2025 Kexing Ying. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kexing Ying
+-/
+import BrownianMotion.StochasticIntegral.UniformIntegrable
+
+/-! # Discrete approximation of a stopping time
+
+-- What kind of indices has `DiscreteApproxSequence`?
+
+-/
+
+open Filter TopologicalSpace
+open scoped NNReal ENNReal Topology
+
+namespace MeasureTheory
+
+variable {ι Ω E : Type*} [TopologicalSpace ι] [TopologicalSpace E]
+  {mΩ : MeasurableSpace Ω} {μ : Measure Ω} {X : ι → Ω → ℝ} {τ : Ω → WithTop ι} {i : ι}
+
+-- Find better name, `RightContinuous` already exists for filtrations
+/-- A stochastic process is right continuous if each of its realizations is right continuous. -/
+abbrev rightContinuous [PartialOrder ι] (X : ι → Ω → E) :=
+  ∀ ω a, ContinuousWithinAt (X · ω) (Set.Ioi a) a
+
+variable [LinearOrder ι] [OrderTopology ι] {𝓕 : Filtration ι mΩ}
+
+/-- Given a random time `τ`, a discrete approximation sequence `τn` of `τ` is a sequence of
+stopping times with countable range that converges to `τ` from above almost surely. -/
+structure DiscreteApproxSequence (𝓕 : Filtration ι mΩ) (μ : Measure Ω := by volume_tac)
+    (τ : Ω → WithTop ι) where
```

Comment:
I don't think we can make a default argument work unless it is in last position. So you should move the measure to be the last argument.